commit 1356d6858058497f222ee43849406e0edc4fa4d1
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-04-30T10:19:39-07:00 (5 months ago)
    
    TRITON-1639 migration failing to provision to a valid server
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/lib/workflows/vm-migration/begin.js b/lib/workflows/vm-migration/begin.js
index dc9ecf1..693a250 100644
--- a/lib/workflows/vm-migration/begin.js
+++ b/lib/workflows/vm-migration/begin.js
@@ -338,7 +338,8 @@ function allocateServer(job, cb) {
 
     // Make sure our VM is placed on a CN away from the old VM.
     job.vmPayload.locality = {
-        far: job.vmPayload.server_uuid
+        far: job.params.vm.uuid,
+        strict: true
     };
 
     // The vmPayload must use vm.vm_uuid (not sure why it's different), anyway,
diff --git a/package.json b/package.json
index 6c30939..f74d152 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.3",
+  "version": "9.8.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/test/lib/migration.js b/test/lib/migration.js
index 566003e..9e59947 100644
--- a/test/lib/migration.js
+++ b/test/lib/migration.js
@@ -1086,8 +1086,8 @@ function TestMigrationCfg(test, cfg) {
     };
 
     test.check_vmapi_state = function test_check_vmapi_state(t) {
-        if (!sourceVm) {
-            t.ok(false, 'Original VM was not created successfully');
+        if (!targetVm) {
+            t.ok(false, 'Vm was not migrated successfully');
             t.done();
             return;
         }
